common_types/src/schema.rs

```diff
@@ -849,12 +881,17 @@ impl TryFrom<common_pb::TableSchema> for Schema {
         let mut builder = Builder::with_capacity(schema.columns.len())
             .version(schema.version)
             .enable_tsid_primary_key(schema.enable_tsid_primary_key);
+        let mut primary_key_index = schema.primary_key_index;
+        if primary_key_index.is_empty() && schema.num_key_columns > 0 {
```

Comment:
No need to keep compatibility, we are still before 1.0, so break change is acceptable now.

Reply:
`num_key_columns` should be removed, and `insert_new_key_column` & `insert_new_normal_column ` could be merged into one method.